File: flutter/hydroponics_framework/lib/datamodels/recipes/system_recipes_view_model.dart
class SystemRecipesViewModel {
  late String plantId;
  late String name;
  late String image;
  late List<String> recipes;

  SystemRecipesViewModel({required this.plantId, required this.name, required this.image, required this.recipes});

  static SystemRecipesViewModel fromJson(Map<String, dynamic> json) =>
      SystemRecipesViewModel(
        plantId : json["plantId"],
        name : json["name"],
        image :json["image"],
        recipes : List<String>.from(json["recipes"])
      );
}

File: kotlin/hydroponics-host/src/main/kotlin/io/greenglass/hydroponics.application/models/recipe/SystemRecipesViewModel.kt
package io.greenglass.hydroponics.application.models.recipe

class SystemRecipesViewModel(
    val plantId : String,
    val name : String,
    val image : String,
    val recipes : List<String>
) {
    constructor(plant : PlantModel, recipes : List<RecipeModel>) : this(
        plant.plantId,
        plant.name,
        plant.image,
        recipes.map { r -> r.recipeId }
    )
}
